addon/pattern-matchers.js

```diff
@@ -51,6 +89,74 @@ class SimplePatternMatchers {
     }
     return false;
   }
+
+  /**
+   * Generates a message indicating why no match was found.
+   * @param {Map<string, string>} testMap
+   * @returns {string}
+   */
+  generateReasonForNoMatching(testMap /* <Map<string, >> */) {
+    const paramNames = [...testMap.keys()].join(SEPARATOR);
+    let reason = `\nbut found no handler for it because\n`;
+
+    let paramNameMatches = [];
+    for (let paramMap of this.patternMapByQueryString.values()) {
+      if ([...paramMap.keys()].join(SEPARATOR) === paramNames) {
+        paramNameMatches.push(paramMap);
+      }
+    }
+
+    let requested = null;
+    let existing = null;
+
+    if (paramNameMatches.length > 0) {
+      requested = this.mapToText(testMap);
+      existing = [];
+      paramNameMatches.forEach((paramMap) =>
+        existing.push(this.mapToText(paramMap))
+      );
+
+      reason = `${reason}query values of:\n`;
+    } else {
+      requested = this.arryToText([...testMap.keys()]);
+      existing = [];
+      for (let paramMap of this.patternMapByQueryString.values()) {
+        existing.push(this.arryToText([...paramMap.keys()]));
+      }
+
+      reason = `${reason}query names of:\n`;
```

Comment:
these are parameter names, or "query parameter names", but not query names

Reply:
Fixed as suggested.